x/staking/src/keeper/unbonding.rs

```diff
@@ -173,63 +188,69 @@ impl<
         if let Some(ref hooks) = self.hooks_keeper {
             hooks.after_validator_begin_unbonding(
                 ctx,
-                validator.get_cons_addr(),
+                validator.cons_addr(),
                 validator.operator_address.clone(),
             );
         }
         Ok(())
     }
 
-    pub fn unbond_all_mature_validators<DB: Database, CTX: TransactionalContext<DB, SK>>(
-        &self,
-        ctx: &mut CTX,
-    ) -> anyhow::Result<()> {
-        // TODO: time in ctx
-        let block_time = Utc::now();
+    pub fn unbond_all_mature_validators<DB: Database>(&self, ctx: &mut BlockContext<'_, DB, SK>) {
+        let block_time = ctx
+            .header()
+            .expect("BlockContext always has header")
+            .time
+            .as_ref()
+            .expect("Expected timestamp in block context header.");
+        let block_time = chrono::DateTime::from_timestamp(block_time.seconds, block_time.nanos as u32)
+            .expect(
+                "Invalid timestamp in transactional header. It means that timestamp contains out-of-range number of seconds and/or invalid nanosecond",
+            );
+
         let block_height = ctx.height() as i64;
 
         // unbondingValIterator will contains all validator addresses indexed under
         // the ValidatorQueueKey prefix. Note, the entire index key is composed as
         // ValidatorQueueKey | timeBzLen (8-byte big endian) | timeBz | heightBz (8-byte big endian),
         // so it may be possible that certain validator addresses that are iterated
         // over are not ready to unbond, so an explicit check is required.
-        let unbonding_val_iterator: HashMap<Vec<u8>, Vec<String>> =
-            self.validator_queue_iterator(ctx, block_time, block_height)?;
+        let unbonding_val_map: HashMap<Vec<u8>, Vec<String>> =
+            self.validator_queue_map(ctx, block_time, block_height);
 
-        for (k, v) in &unbonding_val_iterator {
-            let (time, height) = parse_validator_queue_key(k)?;
+        for (k, v) in &unbonding_val_map {
+            let (time, height) =
+                parse_validator_queue_key(k).expect("failed to parse unbonding key");
 
             // All addresses for the given key have the same unbonding height and time.
             // We only unbond if the height and time are less than the current height
             // and time.
 
             if height < block_height && (time <= block_time) {
                 for addr in v {
-                    let mut validator = self.get_validator(ctx, addr.as_bytes())?;
+                    let val_addr = ValAddress::from_bech32(addr)
+                        .expect("Failed to parse stored ValAddress in validators queue");
+                    let mut validator = self
+                        .validator(ctx, &val_addr)
```

Comment:
The expect calls here don't explain why you think it's safe to unwrap.

Reply:
Added explanation.